### DISCUSSION TITLE
Nested serverless functions

### DISCUSSION THREAD

**Discussion by Nisseboy** [vercel/vercel#8343] [Help] [answered]
I'm using nodejs and i would like to create a serverless function that is run when any subpath is visited
/api/folder/foo/bar
/api/folder/baz.txt
/api/folder/glo/bul/lar

all of these would call the same function (and specify path)

**Comment by amyegan** [comment, marked as answer]
Hi @Nisseboy. It sounds like you might want to use [Path Segments](https://vercel.com/docs/concepts/functions/serverless-functions#path-segments), but I may be misunderstanding.  Can you share a small reproduction of what you're trying to accomplish to show what doesn't work for you? I would be happy to take a closer look once I get more insight into what you want to build.

**Comment by mdubourg001** [comment]
Hi, I'm facing the same issue : I'm trying to have a single Serverless Function (let's say `/api/index.js`) that catches **all incoming requests** :

```
/api/posts                 --> /api/index
/api/posts/42              --> /api/index
/api/absolutely/any/path/  --> /api/index
``` 

[Path Segments](https://vercel.com/docs/concepts/functions/serverless-functions#path-segments) seem to allow this only on one level, and I'm not figuring out how to do this using [Rewrites](https://vercel.com/docs/project-configuration#project-configuration/rewrites) either.

Is it possible ? Do you have any tips how to achieve this ?
Thank you in advance

**Reply by amyegan** [reply]
Hi @mdubourg001! What is possible depends on what framework or library you are using and what you're trying to accomplish. 

Vercel generally only supports the single path segment seen in the doc you shared. But you can use [catch all routes](https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes) in Next.js to handle various numbers path segments to be used as variables in a function. 

The [rewrites](https://vercel.com/docs/project-configuration#project-configuration/rewrites) config option should let you load all matching paths from a single page, but it can be a bit fussy to set up correctly. I prefer to use [middlware](https://vercel.com/docs/concepts/functions/edge-middleware) to handle complex redirects and rewrites.

**Comment by markx** [comment]
Well I'm running into the same issue and my function is written in Go. Guess it's still not supported.
